diagram.md

```diff
@@ -208,11 +208,11 @@ Below is a representation of the Block structure.
 * Data
   * Frame
 
-Although the Timecode value is stored once per Cluster, another timecode is stored within the Block structure itself. The way this works is that the Timecode in the Cluster is relative to the entire `Segment`. It is usually the Timecode that the first Block in the Cluster needs to be played at. The Timecode in the Block itself is relative to the Timecode in the Cluster. For example, let's say that the Timecode in the Cluster is set to 10 seconds, and you have a Block in that Cluster that is supposed to be played 12 seconds into the clip; this means that the Timecode in the Block would be set to 2 seconds.
+The `Timecode Element` value MUST be stored once per `Cluster`. Another `Timecode` SHOULD be stored within the `Block` structure itself. The `Timecode` in the `Cluster` is relative to the entire `Segment`. The `Timecode` SHOULD be the first `Block` in the `Cluster`. The `Timecode` in the `Block` itself is relative to the `Timecode` in the `Cluster`. For example, if the `Timecode` in the `Cluster` is set to 10 seconds and a `Block` in that `Cluster` is supposed to be played 12 seconds into the clip, the `Timecode` in the `Block` would be set to 2 seconds.
```

Comment:
> The `Timecode Element` value MUST be stored once per `Cluster`. Another `Timecode` SHOULD be stored within the `Block` structure itself.

I consider the second use of `Timecode` to be wrong. The first one is talking about the `Timecode Element`, an "unsigned integer" EBML `Element`. The second use is talking about specific parts (bits) of the `Block` element. Yes, it is a timecode, but not a `Timecode` (not an element of its own). Additionally it's actually only an offset to the `Timecode Element` stored in the cluster; adding that offset to the `Timecode Element's` value yields the `Block Element's` actual timecode.

I'd convert this section into something like I've written above, e.g. start with "Each `Cluster` MUST contain a `Timecode` element. Additionally the `Block Element` contains an offset that, when added to the `Cluster's Timecode Element's` value, yields the `Block's` effective timecode."

That offset in the `Block` element isn't optional either; therefore "SHOULD" is definitely the wrong word. Any other qualifier such as MUST or MAY is just as wrong. The `Block` element always consists of several bits that make up said offset. They're a fixture.

> The `Timecode` in the `Block` itself is relative to the `Timecode` in the `Cluster`.

The first use is wrong again as you're talking about those specific bits in the `Block Element` again, and not about a real `EBML Element`.

> …the `Timecode` in the `Block` would be set to 2 seconds.

Dito.